cli_tests/app.sh

```diff
@@ -0,0 +1,172 @@
+#!/usr/bin/env bats
+
+# To run locally:
+# export KETCH_COMMAND=<location of ketch binary>
+# assure you have a kubernetes cluster running w/ traefik, cert manager, etc. (see ketch getting started docs)
+# assure the ketch cli is compiled (make ketch)
+# assure you have bats installed locally (via apt, brew, etc.)
+# ./cli_tests/app.sh
+
+setup() {
+    if [[ -z "${KETCH_COMMAND}" ]]; then
+    KETCH=$(pwd)/bin/ketch
+  else
+    KETCH="${KETCH_COMMAND}"
+  fi
+  INGRESS=$(kubectl get svc traefik -o jsonpath='{.status.loadBalancer.ingress[0].ip}')
+  FRAMEWORK="myframework"
+  APP_IMAGE="docker.io/shipasoftware/bulletinboard:1.0"
+  APP_NAME="bulletinboard"
+  CNAME="my-cname.com"
+  TEST_ENVVAR_KEY="FOO"
+  TEST_ENVVAR_VALUE="BAR"
+}
+
+@test "help" {
+  result="$($KETCH help)"
+  echo "RECEIVED:" $result
+  [[ $result =~ "For details see https://theketch.io" ]]
+  [[ $result =~ "Available Commands" ]]
+  [[ $result =~ "Flags" ]]
+}
+
+@test "framework add" {
+  result=$($KETCH framework add "$FRAMEWORK" --ingress-service-endpoint "$INGRESS" --ingress-type "traefik")
+  echo "RECEIVED:" $result
+  [[ $result =~ "Successfully added!" ]]
+}
+
+@test "framework list" {
+  result=$($KETCH framework list)
+  headerRegex="NAME[ \t]+STATUS[ \t]+NAMESPACE[ \t]+INGRESS TYPE[ \t]+INGRESS CLASS NAME[ \t]+CLUSTER ISSUER[ \t]+APPS"
+  dataRegex="$FRAMEWORK[ \t]+ketch-$FRAMEWORK[ \t]+traefik[ \t]+traefik"
+  echo "RECEIVED:" $result
+  [[ $result =~ $headerRegex ]]
+  [[ $result =~ $dataRegex ]]
+}
+
+@test "framework update" {
+  result=$($KETCH framework update "$FRAMEWORK" --app-quota-limit 1)
+  echo "RECEIVED:" $result
+  [[ $result =~ "Successfully updated!" ]]
+}
+
+@test "app deploy" {
+  run $KETCH app deploy "$APP_NAME" --framework "$FRAMEWORK" -i "$APP_IMAGE"
+  [ $status -eq 0 ]
```

Comment:
can we check the final results output here? Something like "Successfully deployed!" or whatever ketch prints?

Reply:
I only checked the status code here, and for a few other commands, because ketch does not seem to print anything on success. Also, the `app list` test will assert that the app was actually created.